### DISCUSSION TITLE
How do I get infinite scroll working with an ID of a table?

### DISCUSSION THREAD

**Discussion by DavidIlie** [prisma/prisma#18060] [Q&A]
I have two tables, one of them containing a bunch of songs from spotify called `Songs` and another which contains for each user a set of songs to be displayed as a feed called `FeedSong`. My problem is that before using purely songs, I didn't use the ID of the table (didn't put the `@id` there) but instead used its spotify id and everything worked fine. Now since songs can be duplicated for each account, I cannot do that.

These are my two tables:

```prisma
model Song {
  id         String @default(uuid())
  spotify_id String @id
  explicit     Boolean
  name         String
  popularity   Int
  preview_url  String?
  track_number Int
  external_url String
  created     DateTime @default(now())
  lastUpdated DateTime @default(now())
  feeds     FeedSong[]
}

model FeedSong {
  id            String   @id @unique @default(uuid())
  userId        String
  user          User     @relation(fields: [userId], references: [id], onDelete: Cascade)
  songId        String 
  song          Song     @relation(fields: [songId], references: [spotify_id], onDelete: Cascade)
  hasInteracted Boolean  @default(false)
  added         DateTime @default(now())
}
```
I am using TRPC and I can see that in the client it doesn't make the refetch to get new data when it is supposed to, and I saw that it does send back a new cursor which is confusing me.

Here is my API route code:

```ts
const limit = input.limit ?? 15;
      const { cursor, skip } = input;
      const items = (
        await ctx.prisma.feedSong.findMany({
          where: {
            hasInteracted: false,
            userId: ctx.session.user.id,
          },
          take: limit + 1,
          skip: skip,
          cursor: cursor ? { id: cursor } : undefined,
          include: {
            song: {
              include: basicIncludeForSong(ctx.session.user.id),
            },
          },
          orderBy: {
            added: "desc",
          },
        })
      ).map((song) => embedStarAvgToSong<typeof song.song>(song.song));

      let nextCursor: typeof cursor | undefined = undefined;

      console.log(items[0]!.id);
      if (items.length > limit) {
        const nextItem = items.pop();
        console.log(nextItem!.id);
        nextCursor = nextItem!.id;
      }

      return {
        items,
        nextCursor,
      };
```

This is what I have in my frontend:

```ts
  const { data, fetchNextPage, isLoading } = api.feed.getFeed.useInfiniteQuery(
    {
      limit: 15,
    },
    {
      getNextPageParam: (lastPage) => lastPage.nextCursor,
    },
  );
  ```
  
  Any help would be appreciated. Thank you.

**Comment by ludralph** [comment]
Hi @DavidIlie 👋 

Looking at your code, you are using the id field of the FeedSong model as the cursor, but you have defined it as a UUID field that is not sequential, so you cannot use it for cursor-based pagination. Instead, you can use the added field of the FeedSong model as the cursor since it is a DateTime field and should be sortable. See this [section](https://www.prisma.io/docs/concepts/components/prisma-client/pagination#-cons-of-cursor-based-pagination) of the docs.

**Comment by DavidIlie** [comment]
How would I change the cursor to use the `added` field? If I replace this `cursor: cursor ? { id: cursor } : undefined,` to use added it just errors out.

**Reply by ludralph** [reply]
Can you try this 
```js
const limit = input.limit ?? 15;
const { cursor, skip } = input;
const items = await ctx.prisma.feedSong.findMany({
  where: {
    hasInteracted: false,
    userId: ctx.session.user.id,
    added: cursor ? { gt: cursor } : undefined,
  },
  take: limit,
  skip: skip,
  include: {
    song: {
      include: basicIncludeForSong(ctx.session.user.id),
    },
  },
  orderBy: {
    added: "desc",
  },
});

let nextCursor: string | undefined = undefined;

if (items.length > limit) {
  nextCursor = items[limit - 1].added.toISOString();
}

return {
  items,
  nextCursor,
};
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
